### PR TITLE
Fix for #407 Zuul unescaped paths fixed for both Ribbon and SimpleHost routing

### PR DESCRIPTION
**Pull request by brandon-atkinson**
This pull request pulls adds a test to demonstrate the fact that the simple host routing was broken, pulls URI generation into the ProxyRequestHelper, and updates both the SimpleHost and Ribbon routing filters to use the helper.

### CODE REVIEW

File: spring-cloud-netflix-core/src/main/java/org/springframework/cloud/netflix/zuul/filters/route/RibbonCommand.java [hunk 1]
```diff
@@ -80,7 +80,7 @@ public RibbonCommand(String commandKey, RestClient restClient, Verb verb, String
 		super(getSetter(commandKey));
 		this.restClient = restClient;
 		this.verb = verb;
-		this.uri = (StringUtils.hasText(uri))? UriComponentsBuilder.fromUriString(uri).build().toUri() : new URI(uri);
```
Comment (spencergibb) [inline spring-cloud-netflix-core/src/main/java/org/springframework/cloud/netflix/zuul/filters/route/RibbonCommand.java]: I still think the null check needs to happen here.

File: spring-cloud-netflix-core/src/main/java/org/springframework/cloud/netflix/zuul/filters/route/SimpleHostRoutingFilter.java [hunk 2]
```diff
@@ -16,34 +16,12 @@
...
-import java.util.concurrent.atomic.AtomicReference;
-
-import javax.annotation.PreDestroy;
-import javax.net.ssl.SSLContext;
-import javax.net.ssl.TrustManager;
-import javax.net.ssl.X509TrustManager;
-import javax.servlet.http.HttpServletRequest;
-import javax.servlet.http.HttpServletResponse;
-
+import com.netflix.config.DynamicIntProperty;
```
Comment (spencergibb) [inline spring-cloud-netflix-core/src/main/java/org/springframework/cloud/netflix/zuul/filters/route/SimpleHostRoutingFilter.java]: Please update your rules for imports to match ours so no needless changes are made in imports.

File: spring-cloud-netflix-core/src/main/java/org/springframework/cloud/netflix/zuul/filters/route/RibbonCommand.java (hunk 1 above)
Comment (brandon-atkinson) [inline spring-cloud-netflix-core/src/main/java/org/springframework/cloud/netflix/zuul/filters/route/RibbonCommand.java]: The only thing I did here was remove the escaping logic to allow for it to occur outside the ribbon command. The logic in the original would still throw NPE if the RibbonCommand was constructed with a null URI string. 

Does it make more sense to do a null check or make a non-null uri a design constraint with an assertion?

File: spring-cloud-netflix-core/src/main/java/org/springframework/cloud/netflix/zuul/filters/route/SimpleHostRoutingFilter.java (hunk 2 above)
Comment (brandon-atkinson) [inline spring-cloud-netflix-core/src/main/java/org/springframework/cloud/netflix/zuul/filters/route/SimpleHostRoutingFilter.java]: Sorry, I'm not a fan of intellij's pushiness when it comes to import management. Are there any docs that describe the formatting used or do people just do it by trial and error?

File: spring-cloud-netflix-core/src/main/java/org/springframework/cloud/netflix/zuul/filters/route/RibbonCommand.java (hunk 1 above)
Comment (spencergibb) [inline spring-cloud-netflix-core/src/main/java/org/springframework/cloud/netflix/zuul/filters/route/RibbonCommand.java]: My bad, I was thinking the exceptional case was with `URI` if `uri` was empty, but it was with `UriComponentsBuilder`.

File: spring-cloud-netflix-core/src/main/java/org/springframework/cloud/netflix/zuul/filters/route/SimpleHostRoutingFilter.java (hunk 2 above)
Comment (spencergibb) [inline spring-cloud-netflix-core/src/main/java/org/springframework/cloud/netflix/zuul/filters/route/SimpleHostRoutingFilter.java]: We reccomend the [Eclipse Code Formatter Plugin])https://github.com/spring-cloud/spring-cloud-netflix#code-conventions-and-housekeeping) in the docs.  I use this file for imports in that plugin: `https://github.com/spring-cloud/spring-cloud-netflix/blob/master/eclipse/org.eclipse.jdt.ui.prefs`. Generally, if I see the imports are very different, I'll revert those back.

**Comment by spencergibb** [comment]
@brandon-atkinson could you squash your commits?

**Comment by spencergibb** [comment]
Closed via 283c0f7600ffe3bbc41a581534f675f1c829aa33

**Pull request closed without merging**